docs/design/018-new-http-api.md

```diff
@@ -306,6 +306,90 @@ export default async function () {
 }
 ```
 
+#### Events
+
+The new HTTP API will emit events which scripts can subscribe to, in order to implement advanced functionality.
+
+For example, a `requestToBeSent` event is emitted when the request was processed by k6, and just before it is sent to the server. This allows changing the request body or headers, or introducing artificial delays.
+
+```javascript
+import { sleep } from 'k6';
+import { Client } from 'k6/x/net/http';
+
+export default async function () {
+  const client = new Client();
+  client.on('requestToBeSent', event => {
+    console.log(event.type);    // 'requestToBeSent'
+    const request = event.data;
+    request.headers['Cookie'] = 'somecookie=somevalue;'  // overwrites all previously set cookies
+    request.body += ' world!';  // the final body will be 'Hello world!'
+    sleep(Math.random() * 5);   // random delay up to 5s
```

Comment:
nit: let's not use `sleep` in examples. You can use `await delay` and import [this gist](https://gist.githubusercontent.com/mstoykov/7684bf54a1e2f35d048868ba6117e2cb/raw/d44e15a1a942c8aafb9e7a4f7c0d623214b3a651/delay.js) for example. 

This way this examples won't be showing how to block the whole event loop.